gorb/gorb.go

```diff
@@ -0,0 +1,299 @@
+	Method string `json:"method"`
+	Weight int    `json:"weight"`
+	Pulse  Pulse  `json:"pulse"`
+}
+
+// Backend define the backend configuration + service name
+type Backend struct {
+	ServiceName string
+	BackendConf BackendConf
+}
+
+// New creates a gorb handler
+func New(serverBaseURL string, instanceIP string, drainDelay time.Duration, servicesName string, servicesPort string, backendWeight int, backendMethod string, vipLoadbalancer string) (controller.Updater, error) {
+	if serverBaseURL == "" {
+		return nil, errors.New("unable to create Gorb updater: missing server ip address")
+	}
+	initMetrics()
+	log.Infof("Gorb server url: %s, drainDelay: %v, instance ip adddress: %s, vipLoadbalancer: %s ", serverBaseURL, drainDelay, instanceIP, vipLoadbalancer)
+
+	backendDefinition := []Backend{}
+	servicesNameArr := strings.Split(servicesName, ",")
+	servicesPortArr := strings.Split(servicesPort, ",")
+
+	if len(servicesNameArr) != len(servicesPortArr) {
+		return nil, errors.New("Unable to create Gorb updater: the number of serviceName and port are not the same")
+	}
+
+	var backend Backend
+	for index, service := range servicesNameArr {
+
+		port, err := strconv.Atoi(servicesPortArr[index])
+		if err != nil {
+			return nil, errors.New("Unable to convert port form string to int")
+		}
+
+		args := ArgPulse{
+			Method: "GET",
+			Path:   "/",
+			Expect: "404",
+		}
+		pulse := Pulse{
+			Args:            args,
+			TypeHealthcheck: "http",
+			Interval:        "1s",
+		}
+		backend = Backend{
+			ServiceName: service,
+			BackendConf: BackendConf{
+				Host:   instanceIP,
+				Port:   port,
+				Method: backendMethod,
+				Weight: backendWeight,
+				Pulse:  pulse,
+			},
+		}
+
+		backendDefinition = append(backendDefinition, backend)
+	}
+
+	tr := &http.Transport{
+		MaxIdleConns:    10,
+		IdleConnTimeout: 30 * time.Second,
+	}
+	var httpClient = &http.Client{
+		Transport: tr,
+		Timeout:   time.Second * 5,
+	}
+
+	return &gorb{
+		serverBaseURL:   serverBaseURL,
+		drainDelay:      drainDelay,
+		instanceIP:      instanceIP,
+		vipLoadbalancer: vipLoadbalancer,
+		backend:         backendDefinition,
+		httpClient:      httpClient,
+	}, nil
+}
+
+type gorb struct {
+	serverBaseURL   string
+	drainDelay      time.Duration
+	instanceIP      string
+	vipLoadbalancer string
+	httpClient      *http.Client
+	backend         []Backend
+}
+
+func (g *gorb) Start() error {
+	return nil
+}
+
+func (g *gorb) ManageLoopBack(action string, arpIgnore int, arpAnnounce int) error {
+
+	var errorArr *multierror.Error
+	vipLoadbalancerArr := strings.Split(g.vipLoadbalancer, ",")
+	for index, vip := range vipLoadbalancerArr {
+		log.Infof(fmt.Sprintf("VIP %s loopback: %s", action, vip))
+		cmd := fmt.Sprintf("ip addr %s %s/32 dev lo label lo:%d", action, vip, index)
+		_, errCmd := exec.Command("bash", "-c", cmd).Output()
+		errorArr = multierror.Append(errorArr, errCmd)
+
+		log.WithFields(log.Fields{"arp_ignore": arpIgnore}).Info("Set back arp_ignore")
```

Comment:
I'm not sure this log message is accurate.

Reply:
modified